qa-tests-backend/src/test/groovy/ImageSignatureVerificationTest.groovy

```diff
@@ -206,7 +208,11 @@ w9e2Azq1OYIh/pbeBMHARDrBaqqmuMR9+BfAaPAYdkNTU6f58M2zBbuL0A==
     def "Check violations of policy '#policyName' for deployment '#deployment.name'"() {
         expect:
         "Verify deployment has expected violations"
-        assert waitForViolation(deployment.name, policyName, WAIT_FOR_VIOLATION_TIMEOUT) == expectViolations
+        if (expectViolations) {
+            assert waitForViolation(deployment.name, policyName, WAIT_FOR_VIOLATION_TIMEOUT) == expectViolations
+        } else {
+            assert checkForNoViolations(deployment.name, policyName)
```

Comment:
Well, the old code waited for
* 450 seconds in race build,
* otherwise 100 seconds on OpenShift,
* otherwise 30 seconds.

The new code would wait for just 5 seconds.
This may give a false confidence of succeeding test while violations actually appear but after 5 seconds.
Maybe a way to speed things up is by decreasing 450 seconds timeout in race tests? How long does it take for violation to appear when `expectViolations==true` in race tests?

Reply:
Hm, looking again at the test output https://circleci.com/api/v1.1/project/github/stackrox/stackrox/575517/output/134/0?file=true&allocation-id=6285e4d6387d7e6780308a30-0-build%2F38D39C05

`expectViolations==true` cases find violations... after `0` seconds. Checks for no-violations are using up full timeout.

```
ImageSignatureVerificationTest > Check violations of policy 'Distroless' for deployment 'with-signature-verified-by-tekton' STANDARD_OUT
    07:31:16 | INFO  | ImageSignatureVerificationTest | Starting testcase
    07:31:17 | INFO  | Services                  | violation size is: 1
    07:31:17 | INFO  | Services                  | Distroless triggered after waiting 0 seconds
    07:31:17 | INFO  | ImageSignatureVerificationTest | Ending testcase

ImageSignatureVerificationTest > Check violations of policy 'Distroless' for deployment 'with-signature-unverifiable' STANDARD_OUT
    07:31:17 | INFO  | ImageSignatureVerificationTest | Starting testcase
    07:31:18 | INFO  | Services                  | violation size is: 1
    07:31:18 | INFO  | Services                  | Distroless triggered after waiting 0 seconds
    07:31:18 | INFO  | ImageSignatureVerificationTest | Ending testcase

ImageSignatureVerificationTest > Check violations of policy 'Distroless' for deployment 'without-signature' STANDARD_OUT
    07:31:18 | INFO  | ImageSignatureVerificationTest | Starting testcase
    07:31:18 | INFO  | Services                  | violation size is: 1
    07:31:18 | INFO  | Services                  | Distroless triggered after waiting 0 seconds
    07:31:18 | INFO  | ImageSignatureVerificationTest | Ending testcase

ImageSignatureVerificationTest > Check violations of policy 'Distroless' for deployment 'with-signature-verified-by-distroless' STANDARD_OUT
    07:31:18 | INFO  | ImageSignatureVerificationTest | Starting testcase
    07:38:54 | INFO  | Services                  | Failed to trigger Distroless after waiting 456 seconds
    07:38:54 | INFO  | ImageSignatureVerificationTest | Ending testcase
```

I suggest to tighten up timeouts the following way:
1. Use explicitly small timeout in the call to `waitForViolation()`.
2. Use timeout three times as big in the call to `checkForNoViolations()`.

Both items together give a confidence that when the test that expects violation passes and they appear in time `T` (or less), the succeeding test that expects no violations to appear would also legitimately pass because `3*T` should be more than enough for any violations to appear and fail it.

Reply:
> Checks for no-violations are using up full timeout.

I'd say this is expected, as the code behind `checkForNoViolations()` will always wait the full specified timeout.

I adjusted the timeout values to now use 15 seconds for both, which will add an overhead of 90 seconds. Have to carefully observe whether this will introduce any unwanted flakes in the future.